src/data_structures/suffix_array.rs

```diff
@@ -19,12 +20,79 @@ use vec_map::VecMap;
 
 use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
+use data_structures::bwt::{BWT, Less, Occ};
 
+pub trait SuffixArray {
+    fn get(&self, index: usize) -> Option<usize>;
+    fn range(&self, range: Range<usize>) -> Option<Vec<usize>>;
```

Comment:
The error is this: 
```
src/data_structures/suffix_array.rs:105:21: 105:42 error: the trait `core::ops::Deref` is not implemented for the type `[usize]` [E0277]
src/data_structures/suffix_array.rs:105                Some(Box::new(self[range]))
                                                            ^~~~~~~~~~~~~~~~~~~~~
src/data_structures/suffix_array.rs:105:21: 105:42 help: run `rustc --explain E0277` to see a detailed explanation
src/data_structures/suffix_array.rs:105:21: 105:42 note: required for the cast to the object type `core::ops::Deref<Target=[usize]>`
src/data_structures/suffix_array.rs:105:21: 105:42 error: the trait `core::marker::Sized` is not implemented for the type `[usize]` [E0277]
src/data_structures/suffix_array.rs:105                Some(Box::new(self[range]))
                                                            ^~~~~~~~~~~~~~~~~~~~~
src/data_structures/suffix_array.rs:105:21: 105:42 help: run `rustc --explain E0277` to see a detailed explanation
src/data_structures/suffix_array.rs:105:21: 105:42 note: `[usize]` does not have a constant size known at compile-time
src/data_structures/suffix_array.rs:105:21: 105:42 note: required for the cast to the object type `core::ops::Deref<Target=[usize]>`
src/data_structures/suffix_array.rs:105:21: 105:29 error: the trait `core::marker::Sized` is not implemented for the type `[usize]` [E0277]
src/data_structures/suffix_array.rs:105                Some(Box::new(self[range]))
                                                            ^~~~~~~~
```
The issue is that for `Some(Box::new(self[range]))` to be an `Option<Box<Deref<Target = [usize]>>>`, `self[range] ` needs to be `Deref<Target=[usize]>`, which it isn't.

In order to fix this, I tried `Some(Box::new(Box::new(self[range])))` and several variants thereof, all of which gave me the error that `[usize]` is unsized, so I can't `Box` it (`Box` can abstractly hold unsized types such as traits, but the size has to be known at the time of creation, I believe).